tests/queries/tests.py

```diff
@@ -3797,3 +3798,97 @@ def test_ticket_23622(self):
             set(Ticket23605A.objects.filter(qy).values_list('pk', flat=True))
         )
         self.assertSequenceEqual(Ticket23605A.objects.filter(qx), [a2])
+
+
+@skipUnlessDBFeature('supports_select_union')
```

Comment:
I suggest putting this in a separate file since 3.8K lines is already a bit large (there's a ticket to refactor this file, if possible).